polynote-kernel/src/main/scala/polynote/kernel/interpreter/python/PythonInterpreter.scala

```diff
@@ -244,12 +244,137 @@ class PythonInterpreter private[python] (
       |from pathlib import Path
       |from java.lang import RuntimeException, StackTraceElement
       |from java.util import ArrayList, HashMap
+      |from importlib.util import spec_from_loader
+      |from collections import defaultdict, UserDict
       |from polynote.kernel import Pos
       |from polynote.kernel import KernelReport
       |
       |if not hasattr(sys, 'argv') or len(sys.argv) == 0:
       |    sys.argv  = ['']
       |
+      |class DelegatingFinder(object):
```

Comment:
I wonder if it's time to start thinking about moving this to a python source file somehow... 😄

Reply:
😬

Reply:
Yeah, we're building up quite a lot of logic in these strings, would be good to extract them but:

1. Really don't want to deal with python packaging if I can avoid it 😟 
2. A bit iffy about the jep interop stuff. 

We will have to bite the bullet eventually though